crates/nu-command/src/network/http/head.rs

```diff
@@ -0,0 +1,178 @@
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, SyntaxShape, Type, Value,
+};
+
+use crate::network::http::client::{
+    http_client, http_parse_url, request_add_authorization_header, request_add_custom_headers,
+    request_handle_response, request_set_timeout,
+};
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "http head"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("http head")
+            .input_output_types(vec![(Type::Nothing, Type::Any)])
+            .allow_variants_without_examples(true)
+            .required(
+                "URL",
+                SyntaxShape::String,
+                "the URL to fetch the contents from",
+            )
+            .named(
+                "user",
+                SyntaxShape::Any,
+                "the username when authenticating",
+                Some('u'),
+            )
+            .named(
+                "password",
+                SyntaxShape::Any,
+                "the password when authenticating",
+                Some('p'),
+            )
+            .named(
+                "max-time",
+                SyntaxShape::Int,
+                "timeout period in seconds",
+                Some('m'),
+            )
+            .named(
+                "headers",
+                SyntaxShape::Any,
+                "custom headers you want to add ",
+                Some('H'),
+            )
+            .switch(
+                "raw",
+                "fetch contents as text rather than a table",
+                Some('r'),
+            )
+            .switch(
+                "insecure",
+                "allow insecure server connections when using SSL",
+                Some('k'),
+            )
+            .filter()
+            .category(Category::Network)
+    }
+
+    fn usage(&self) -> &str {
+        "Get the headers from a URL."
+    }
+
+    fn extra_usage(&self) -> &str {
+        "Performs HTTP HEAD operation."
+    }
+
+    fn search_terms(&self) -> Vec<&str> {
+        vec!["network", "request", "curl", "wget", "headers", "header"]
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        run_head(engine_state, stack, call, input)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![
+            Example {
+                description: "Get headers from example.com",
+                example: "http head https://www.example.com",
```

Comment:
Is this supposed to work? I'm trying to figure out how to test whether http head is working.

Reply:
You're right, HTTP HEAD is not as useful as the others. The problem is, unless, we don't have access to headers response (from a user perspective), this command will not be as useful as the others. Since the only output is the headers. And as for now, we only display the body if it exists. I know there was ongoing discussion to better expose the output of HTTP commands, but I don't know if the person involved did any progress on it.

Reply:
![image](https://user-images.githubusercontent.com/31616285/220766362-ccf3c8d3-113f-4ba7-b06d-0bd36e6c80a0.png)

Head is supposed to work in the same way as get, except you won't receive the payload, just the headers.

As of today, those headers are not exposed to the user. The HTTP commands do only return the payload. I think we should update the output of those commands to return at least the headers + the payload.

Reply:
> Head is supposed to work in the same way as get, except you won't receive the payload, just the headers.

Then I think it would be nice to have a tabular output of those headers as the output of this command. What do you think? Am I out of line?

Reply:
No, I was just wondering how to do it in an extendable way. But having a list here is also possible. I'll try to work on it today.

Reply:
oh, wow! That's very cool! Great job!!!